Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/core/execution/VaccinationUpdateScheduler.kt

```diff
@@ -0,0 +1,116 @@
+package de.rki.coronawarnapp.vaccination.core.execution
+
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.WorkInfo
+import androidx.work.WorkManager
+import de.rki.coronawarnapp.task.TaskController
+import de.rki.coronawarnapp.task.TaskFactory
+import de.rki.coronawarnapp.task.common.DefaultTaskRequest
+import de.rki.coronawarnapp.util.TimeAndDateExtensions.toLocalDateUtc
+import de.rki.coronawarnapp.util.TimeStamper
+import de.rki.coronawarnapp.util.coroutine.AppScope
+import de.rki.coronawarnapp.util.coroutine.await
+import de.rki.coronawarnapp.util.device.ForegroundState
+import de.rki.coronawarnapp.vaccination.core.execution.task.VaccinationUpdateTask
+import de.rki.coronawarnapp.vaccination.core.execution.worker.VaccinationUpdateWorkerRequestBuilder
+import de.rki.coronawarnapp.vaccination.core.repository.VaccinationRepository
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.onEach
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class VaccinationUpdateScheduler @Inject constructor(
+    @AppScope private val appScope: CoroutineScope,
+    private val taskController: TaskController,
+    private val vaccinationRepository: VaccinationRepository,
+    private val foregroundState: ForegroundState,
+    private val workManager: WorkManager,
+    private val workerRequestBuilder: VaccinationUpdateWorkerRequestBuilder,
+    private val timeStamper: TimeStamper,
+) {
+
+    fun setup() {
+        Timber.tag(TAG).d("setup()")
+
+        // If there is a pending check, we can perform it in the background.
+        // We basically consume all "pending check flags" in the background until there are none.
+        vaccinationRepository.vaccinationInfos
+            .map { vaccinatedPersons ->
+                vaccinatedPersons.any { it.isProofCertificateCheckPending }
+            }
+            .distinctUntilChanged()
+            .onEach { hasProofCheckPending ->
```

Comment:
The above `.map` checks whether any person in our data set has a pending proofCheck.
Via the `distinctUntilChanged` we make sure to only execute the code if `hasProofCheckPending` actually changes (though it would not be an issue because we scheduler the worker with `KEEP`).
So the `onEach { hasProofCheckPending: Boolean ->` ` is already the combination of every person.